HAL.md

```diff
@@ -0,0 +1,110 @@
+# Hardware abstract layer(HAL) overview
+
+## Abstract
+Hardware abstract layer(HAL) defines a standard interface for hardware vendors to implement, which enables Tangle-Accelerator to be aware of lower-level driver implementations. HAL layer supports such as uart, GPIO, secure storage and other low-level operators inside endpoint. The endpoint is a daemon, offering monitor for the device and being able to send the message to tangle-accelerator. HAL layer lets the user only need to implement their device operators from abstract interface which is defined inside `device.h` but not need to know how endpoint works.
+
+## Directory overview
+* connectivity: Functions that communicate to tangle-accelerator.
+* devices: The real directory to implement devices.
+* hal: Hardware abstraction layer.
+* hooks: Git hook functions.
+* tests: Test functions and unit-tests.
+* third_party: Third party directory.
+* utils: Utility functions for TA-endpoint.
```

Comment:
Think of the merging process between TA and TA-endpoint. It is not necessary to make a list of directories at the moment. Instead, you can send further pull requests after this one is merged.

Reply:
HAL is a blocking issue for merge process though.